lib/matplotlib/text.py

```diff
@@ -107,7 +107,6 @@ class Text(Artist):
     """Handle storing and drawing of text in window or data coordinates."""
 
     zorder = 3
-    _cached = cbook.maxdict(50)
```

Comment:
Is this dict really a memory hog if it gets larger?  Why not just set to a large number and move on?  Just because we  can make a double caching structure doesn't mean we should.  Given that we have had very few complaints of this nature, my assumption is that most people are not hitting the 50-element limit, so bumping it to 5000 for the few people who need it doesn't seem like a terrible idea.   

Note that this largely catches users who have accidentally used categoricals and created thousands of ticks.  I think, if anything, we should add logic in the categorical ticker that raises a warning of more than 100 ticks are being asked for.